### DISCUSSION TITLE
Briggs & Stratton / GE Model GC1030 Controller Connections

### DISCUSSION THREAD

**Discussion by gfvillano** [jgyates/genmon#1160] [Questions]
### Checklist

- [X] I made sure that there are *no existing issues or discussions* - [open issues](https://github.com/jgyates/genmon/issues),[closed issues](https://github.com/jgyates/genmon/issues?q=is%3Aissue+is%3Aclosed),[open discussion](https://github.com/jgyates/genmon/discussions), [closed discussion](https://github.com/jgyates/genmon/discussions/categories/questions?discussions_q=is%3Aclosed) - which may answer my question or that I could contribute my information to.
- [X] I have read the [Known Issues Wiki Page](https://github.com/jgyates/genmon/wiki/Appendix-D-Known-Issues) and my question / issue is not listed.
- [X] I have provided the version of the software and the operating system in my post.
- [X] If I have an issue with genmon not communicating with the controller, I have read and followed the [serial troubleshooting guide](https://github.com/jgyates/genmon/wiki/3.6---Serial-Troubleshooting).
- [X] I have included the generator make and model in my submission.

### Version

Not installed yet

### Have you submitted your log files from the About page? Note that outbound email must be working for logs to be submitted.

I have not submitted logs

### What is/are your question(s)?

Currently working to determine the Modbus communication wiring for my B&S/GE 20 KW Model 40334 generator that has the GC1030 controller. This unit originally had the GE InfoHub system installed )and working great for years) until the 2G cell service ceased to exist three years ago. The interface board that was a part in the InfoHub kit (to split / share the data signals to the hub) is still in place and I was hoping to leverage that cabling (five wires - red +12DVC, black ground, grey +12VDC, brown data signal?, yellow data signal?) to connect directly to a PUSR USR-W630. Once that is in place, I will connect the Pi via wifi to the W630. Hoping to verify the connections to simplify the steps. Thank you!

### Additional information

![Screenshot 2024-07-29 at 2 36 46 PM](https://github.com/user-attachments/assets/4f8ea4df-c5e1-48b7-803c-fd25ec966f5f)
![IMG_2407](https://github.com/user-attachments/assets/bc2ec034-9e0b-4e68-bf50-313cc68fdc3f)
![IMG_2411](https://github.com/user-attachments/assets/b470e76d-8cbd-4fbe-b37f-1554263bb7d3)
![IMG_2418](https://github.com/user-attachments/assets/2027b3e3-ff8d-40ea-82be-30b53f5bf252)

**Comment by jgyates** [comment]
Do you know if this is RS-485 or RS-232? You could possibly tell the TX and RX by voltage levels. Receive would be Hi-Z or floating, TX would be a negative voltage in the -12V to -3V range, RS-485 would be two wires and the difference between them would be Hi-Z or floating, possible with a resistance of in the ball park of 100 ohms.

**Reply by gfvillano** [reply]
I am not certain if the signals are RS232 or RS485. I had connected a Saleae Logic 4 logic analyzer on it earlier today with zero results (from chassis ground to the yellow wire on one channel and chassis ground to the brown wire on another channel). I may try the Saleae again but use the grey wire as the common point as the logic analyzer is protected up to 25V. I have been searching the web for days trying to find the details on the interface board outputs with no success.

**Reply by skipfire** [reply]
Page 45 has a diagram that indicates RS485 for their WiFi, not sure if that helps any or not.  https://www.electricgeneratorsdirect.com/manuals/1a2c67d5b578d1b24297cf834820d944.pdf

**Reply by gfvillano** [reply]
Appreciate that info.  This controller is ten years old and two versions ahead of the 1032 controller so the terminal strip connections in the field wiring box are shown below - no wifi or cell terminals labeled. But the cellular InfoHub unit shared a fair amount of data so it's a matter of determining what specifics are around the protocol and settings. The transfer panel is only a two wire comms cable though. 
![Screenshot 2024-07-29 at 8 09 16 PM](https://github.com/user-attachments/assets/3cd2f6c6-4d2e-40d2-a40d-bd1fed4a8f13)
![Screenshot 2024-07-29 at 8 08 56 PM](https://github.com/user-attachments/assets/1a7ee345-3bec-48e5-ad55-53d1939354a5)
I just tried several different variations of baud rates with the analyzer - still nothing though. I was able to get the Pi running at least.

**Reply by jgyates** [reply]
The project wiki has a page for the Briggs and Stratton GC-1030:

https://github.com/jgyates/genmon/wiki/Appendix-P-Briggs-and-Stratton-Controller-Information

I think the data rate is set in the controller menus.

**Reply by gfvillano** [reply]
This 1030 controller does not have the functionality that the 1031 & 1032 have to access those types of lower level / service menus from the controller keypad. I may need to try to get my hands on the old software and programming cable to get that deep into the system. Let me know if anyone may have access to someone that could sell me that outdated and probably unsupported software.

**Reply by jgyates** [reply]
you could just try the following data rates: 9600, 19200, 38400, 57600, 115200. Chances are it is one of these.
